### PR TITLE
Added some convenience methods to MaskData

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

This PR fixes one bug in the MaskData class (line 316), adds some convenience methods which allow setting the mask using a binary mask (as int or boolean array) and adds some unit tests.

I found the MaskData class very difficult to use as it is, hence this PR. I think in most circumstances one simply wants to create a mask from some kind of binary array (either boolean or int).
Does that make sense? If that's not the case, I'm happy to close the PR again.

# Testing this PR

Check that the unit tests make sense and pass.

### CODE REVIEW

File: components/blitz/src/omero/gateway/model/MaskData.java [hunk 1]
```diff
@@ -368,5 +368,198 @@ public byte getBit(byte[] data, int bit)
...
+    /**
+     * Set the mask; automatically crop to its bounding box.
+     * (sets X, Y, width and height accordingly)
+     * Needs the height initially set to the image height.
+     * 
+     * @param mask
+     *            The binary mask covering the whole
+     *            image
+     */
+    public void setMaskAndCrop(int[] mask) {
```
Comment (dominikl) [inline components/blitz/src/omero/gateway/model/MaskData.java]: Thought about it again... these methods should better go into some kind of utility class not in the MaskData itself. Will remove the `setMaskAndCrop` methods again.

File: components/blitz/src/omero/gateway/model/MaskData.java [hunk 2]
```diff
@@ -368,5 +369,140 @@ public byte getBit(byte[] data, int bit)
...
+     * [4 5 6 7]
+     * [8 9 f f]
+     * 
+     * @param input A boolean array
+     * @param length The length of the chunks
+     * @return Two dimensional array
+     */
+    private boolean[][] fold(boolean[] input, int length) {
+        int height = input.length / length;
+        if ( input.length % length != 0)
```
Comment (jburel) [inline components/blitz/src/omero/gateway/model/MaskData.java]: extra space before bracket

File: components/blitz/src/omero/gateway/model/MaskData.java [hunk 3]
```diff
@@ -368,5 +369,140 @@ public byte getBit(byte[] data, int bit)
...
 
+    /**
+     * Transforms an integer array to a boolean array,
+     * where 0 == false and !0 == true
+     * @param array The integer array
+     * @return The boolean array
+     */
+    private boolean[][] intToBoolean(int[][] array) {
+        boolean[][] result = new boolean[array.length][array[0].length];
+        for (int i=0; i<array.length; i++)
```
Comment (jburel) [inline components/blitz/src/omero/gateway/model/MaskData.java:462]: is it possible to unify a bit cf. ``for (int i = 0; i < input.length; i++)``

File: components/blitz/src/omero/gateway/util/Mask.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        
+        int width = mask.length;
+        int height = mask[0].length;
+        
+        for (int y = height-1 ; y >= 0 ; y--)
+        {
+            for (int x = 0 ; x < width ; x++)
+            {
+                if (mask[x][y]) {
+                    if ( x < minx )
```
Comment (jburel) [inline components/blitz/src/omero/gateway/util/Mask.java:73]: can you unify sometime no space after bracket some space

File: components/blitz/test/omero/model/MaskDataTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        mask.setHeight(h);
+        mask.setWidth(w);
+        
+        mask.setMask(binMask);
+        
+        int[][] got = mask.getMaskAsBinaryArray();
+        Assert.assertEquals(got.length, 10);
+        Assert.assertEquals(got[0].length, 15);
+        for (int i=0; i<binMask.length; i++) {
+            for(int j=0; j<binMask[i].length; j++) {
```
Comment (jburel) [inline components/blitz/test/omero/model/MaskDataTest.java]: unify

File: components/blitz/test/omero/model/MaskDataTest.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            }
+        }
+        
+        mask.setMask(binMask2);
+        
+        got = mask.getMaskAsBinaryArray();
+        Assert.assertEquals(got.length, 10);
+        Assert.assertEquals(got[0].length, 15);
+        for (int i=0; i<binMask.length; i++) {
+            for(int j=0; j<binMask[i].length; j++) {
```
Comment (jburel) [inline components/blitz/test/omero/model/MaskDataTest.java]: same

**Comment by jburel** [comment]
Are the tests run? I cannot see them in the output

**Comment by dominikl** [comment]
I'm pretty sure the tests were run from the CI in the past... but can't find the output now either.

**Comment by dominikl** [comment]
Now the unit tests run again! https://api.travis-ci.org/v3/job/455408658/log.txt

**Comment by jburel** [comment]
The unit tests are now passing
```
PASSED: testCreateCroppedMask
[nomemorytestng] PASSED: testCreateCroppedMasks
```

```
[nomemorytestng] PASSED: testSetGetBit
[nomemorytestng] PASSED: testSetMaskAsBoolean
[nomemorytestng] PASSED: testSetMaskAsInt
```

**Pull request merged**